src/Sparrow/Threading/LightWeightThreadLocal.cs

```diff
@@ -11,13 +11,27 @@ public sealed class LightWeightThreadLocal<T> : IDisposable
         [ThreadStatic]
         private static CurrentThreadState _state;
 
-        private ConcurrentDictionary<CurrentThreadState, T> _values = new ConcurrentDictionary<CurrentThreadState, T>(ReferenceEqualityComparer<CurrentThreadState>.Default);
+        private readonly WeakReferenceCompareValue<LightWeightThreadLocal<T>> SelfReference;
+        private ConcurrentDictionary<WeakReferenceCompareValue<CurrentThreadState>, T> _values = new ConcurrentDictionary<WeakReferenceCompareValue<CurrentThreadState>, T>();
         private readonly Func<T> _generator;
         private readonly GlobalState _globalState = new GlobalState();
 
         public LightWeightThreadLocal(Func<T> generator = null)
         {
             _generator = generator;
+            SelfReference = new WeakReferenceCompareValue<LightWeightThreadLocal<T>>(this);
+        }
+
+        public List<object> GetTargets()
```

Comment:
Is it for debugging? Can we change accessibility to `internal`?

Reply:
Removed